src/httpserver/http_utils.hpp

```diff
@@ -299,6 +301,7 @@ class header_comparator {
 **/
 class arg_comparator {
  public:
+     using is_transparent = std::true_type;
```

Comment:
What is this doing? (I cannot find it used elsewhere)

Reply:
This is required to activate heterogeneous lookup using this comparator. `is_transparent` just needs to be defined, it doesn't get referenced anywhere. From cppreference:

The member type is_transparent indicates to the caller that this function object is a transparent function object: it accepts arguments of arbitrary types and uses perfect forwarding, which avoids unnecessary copying and conversion when the function object is used in heterogeneous context, or with rvalue arguments. In particular, template functions such as [std::set::find](https://en.cppreference.com/w/cpp/container/set/find) and [std::set::lower_bound](https://en.cppreference.com/w/cpp/container/set/lower_bound) make use of this member type on their Compare types.